components/common/src/main/kotlin/com/hotels/styx/metrics/CentralisedMetrics.kt

```diff
@@ -0,0 +1,326 @@
+/*
+  Copyright (C) 2013-2021 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.metrics
+
+import com.hotels.styx.api.HttpResponseStatus
+import com.hotels.styx.api.extension.Origin
+import com.hotels.styx.api.metrics.MeterFactory
+import com.hotels.styx.common.SimpleCache
+import io.micrometer.core.instrument.*
+
+/**
+ * All the metrics used in Styx are defined here. Please note that there may be additional metrics defined by any plugins used.
+ *
+ * ## General information
+ *
+ * Metrics that do not require tags are registered here, and accessible as values. For convenience when accessing from Java code,
+ * the annotation @get:JvmName(...) is used, which defines a method name by which the value may be accessed.
+ * For consistency, the name of the value and method must be the same.
+ *
+ * Metrics that do require tags are not registered here, but instead provide methods for registering them, with appropriate parameters.
+ * If these parameters are not dependent upon configuration, the metrics may be cached here.
+ * * Examples of dependent on configuration: app ID, origin ID, plugin name.
+ * * Example of not-dependent on configuration (a.k.a. internal behaviour): response status code, request cancellation cause.
+ *
+ * ## Gauges
+ * Gauges cannot be pre-registered here, as their registration requires a callback. Instead we implement an interface called "GaugeId",
+ * which provides the ability to register the gauge, but not to change the name or tags after they have been defined.
+ *
+ * ## Counters
+ * As counters are quite straightforward, we define them using the micrometer Counter class directly.
+ *
+ * ## Timers
+ * For timers we use our own interface "TimerMetric" instead of the Timer interface from micrometer. This is to make the API more
+ * intuitive to work with, and improve encapsulation.
+ */
+class CentralisedMetrics(val registry: MeterRegistry) {
+    /* ------ GAUGES ------ */
+
+    /**
+     * Measures the number of requests that have been received, but not yet fully responded to.
+     */
+    @get:JvmName("requestsInProgress")
+    val requestsInProgress: GaugeId = InnerGaugeId("proxy.request.outstanding", Tags.empty())
```

Comment:
Can we not use `proxy` prefix? It confused me at the beginning when I was migrating HCOM dashboards.